pkg/state/assets.go

```diff
@@ -23,70 +24,72 @@ type assetInfo struct {
 	assetChangeableInfo
 }
 
-func (ai *assetInfo) isNFT() bool {
-	return ai.quantity.Uint64() == 1 && ai.decimals == 0 && !ai.reissuable
+func (ai *assetInfo) initIsNFTFlag(fs featuresState) error {
+	if !ai.quantity.IsUint64() {
+		ai.isNFT = false
+		return nil
+	}
+	ap := assetParams{ai.quantity.Uint64(), uint32(ai.decimals), ai.reissuable}
+	nft, err := isNFT(fs, ap)
+	if err != nil {
+		return err
+	}
+	ai.isNFT = nft
+	return nil
 }
 
 func (ai *assetInfo) equal(ai1 *assetInfo) bool {
 	return ai.assetChangeableInfo.equal(&ai1.assetChangeableInfo) && (ai.assetConstInfo == ai1.assetConstInfo)
 }
 
-// assetConstInfoSize = len(digestTail) + len(issuerPK) + len(decimals_byte) + len(issueHeight_bytes)
-const assetConstInfoSize = proto.AssetIDTailSize + crypto.PublicKeySize + 1 + 8 + 8
-
 // assetConstInfo is part of asset info which is constant.
 type assetConstInfo struct {
 	tail                 [proto.AssetIDTailSize]byte
 	issuer               crypto.PublicKey
 	decimals             uint8
 	issueHeight          proto.Height
+	isNFT                bool // isNFT flag must be initialized only once by calling assetInfo.initIsNFTFlag
 	issueSequenceInBlock uint32
 }
 
+type assetConstInfoRecord struct {
```

Comment:
Why keep both types `assetConstInfo` and `assetConstInfoRecord`? May be remove one? Or convert the old one in CBOR record?